src/plugins/intel_gna/src/transformations/transpose_2d.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/transpose_2d.hpp"
+
+#include "common/graph_utils.hpp"
+#include "openvino/cc/ngraph/itt.hpp"
+#include "openvino/opsets/opset11.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+
+using namespace ov::opset11;
+using namespace ov::pass;
+using namespace ov::intel_gna::pass;
+
+Transpose2D::Transpose2D() {
+    MATCHER_SCOPE(Transpose2D);
+
+    auto transpose_const = pattern::wrap_type<Constant>();
+    auto transpose =
+        pattern::wrap_type<Transpose>({pattern::any_input(), transpose_const}, [](const ov::Output<ov::Node>& node) {
+            return (graph_utils::trim_shape(node.get_shape()).size() > 2);
+        });
+
+    ov::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        const auto transpose_order = as_type_ptr<Constant>(pattern_map.at(transpose_const).get_node_shared_ptr());
+        const auto transpose_node = pattern_map.at(transpose).get_node_shared_ptr();
+        const ov::Shape& shape = transpose_node->get_input_shape(0);
+        const ov::Shape& shape_out = transpose_node->get_output_shape(0);
+        ov::AxisVector axises = transpose_order->get_axis_vector_val();
+        ov::AxisVector axises_fused = {};
+        ov::Shape shape_fused_out = {};
```

Comment:
I think we should not use word "fuse" in the context of this transformation, it may be confusing (pun intended ;-) ). We use fusing for whole layers. I suggest squeeze, although maybe it's not perfect.